jujugui/static/gui/src/app/models/models.js

```diff
@@ -2343,22 +2343,29 @@ YUI.add('juju-models', function(Y) {
 
        @method _mapServicesToMachines
        @param {Object} machineList The list of machines.
+       @param {Boolean} includeUncommitted whether to include uncommitted
+         machines.
      */
-    _mapServicesToMachines: function(machineList) {
+    _mapServicesToMachines: function(machineList, includeUncommitted) {
       var machinePlacement = {};
       var owners = {};
       var machineNames = {};
+      let machines;
 
-      // Strip uncommitted machines and containers from the machine list.
-      var db = this; // machineList.filter does not respect bindscope.
-      var machines = machineList.filter(function(machine) {
-        if (machine.commitStatus === 'uncommitted' ||
-            machine.commitStatus === 'in-progress' ||
-            db.units.filterByMachine(machine.id).length === 0) {
-          return false;
-        }
-        return true;
-      });
+      if (includeUncommitted) {
+        machines = machineList.toArray();
+      } else {
+        // Strip uncommitted machines and containers from the machine list.
+        var db = this; // machineList.filter does not respect bindscope.
```

Comment:
You could do this a couple ways. Because this is YUI the `filter` function takes a second argument which is the execution context. The other (and preferred) is to use an error function :)